packages/backend/src/tokens/authenticateContext.ts

```diff
@@ -0,0 +1,88 @@
+import { constants } from '../constants';
+import type { ClerkRequest } from './clerkRequest';
+import type { AuthenticateRequestOptions } from './types';
+
+interface AuthenticateContextInterface extends AuthenticateRequestOptions {
+  // header-based values
+  sessionTokenInHeader: string | undefined;
+  origin: string | undefined;
+  host: string | undefined;
+  forwardedHost: string | undefined;
+  forwardedProto: string | undefined;
+  referrer: string | undefined;
+  userAgent: string | undefined;
+  secFetchDest: string | undefined;
+  accept: string | undefined;
+  // cookie-based values
+  sessionTokenInCookie: string | undefined;
+  clientUat: number;
+  // handshake-related values
+  devBrowserToken: string | undefined;
+  handshakeToken: string | undefined;
+  // url derived from headers
+  clerkUrl: URL;
+  // cookie or header session token
+  sessionToken: string | undefined;
+}
+
+interface AuthenticateContext extends AuthenticateContextInterface {}
```

Comment:
❓ Why are we doing this?

Reply:
We're merging the interface with the class so we don't have to redefine the interface's props in the class it self

Reply:
Makes sense :+1:. In that case, I believe we can get rid of the intermediary `AuthenticateContextInterface` type ([TS Playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgPIAczAPYgM7IDeAUMsjNtgFzJ5hSgDmxAvscaJLIigMK6QAHmGQRhEEABMCGLLgIkyAIzhQadBiGZtiCADZw8BfuDEjFyFQC8AFAEoipMsigQwAVyghkYABbA8ADoVKCc2HQR5EQRkAF5kEAgAd2QTITB7dgRAoA)). 

This should work:

```typescript
interface AuthenticateContext extends AuthenticateRequestOptions {
// all the properties defined in AuthenticateContextInterface above...
}

class AuthenticateContext {
// ...
}
```